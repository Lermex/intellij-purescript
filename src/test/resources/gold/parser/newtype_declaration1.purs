module A where

newtype User1
