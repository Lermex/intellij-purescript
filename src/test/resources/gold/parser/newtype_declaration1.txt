Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User1')
        PSASTWrapperElement(TypeArgs)
          <empty list>