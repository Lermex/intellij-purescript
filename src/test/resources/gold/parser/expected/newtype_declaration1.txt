Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
        PsiElement(PSTokenType.newtype)('newtype')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User1')